File: tests/code/corpus/langs/csharp/Registry.cs
using System.Collections.Generic;

namespace Geometry
{
    public class Registry
    {
        private readonly List<object> items = new List<object>();

        public Registry() { }

        public void Add(object item)
        {
            items.Add(item);
        }

        public void Seed()
        {
            Add(Factory.MakeCircle(1.0));
        }
    }
}

File: tests/code/corpus/langs/scala/Registry.scala
package geometry

import scala.collection.mutable.ListBuffer

class Registry {
  private val items = ListBuffer[Any]()
  def add(item: Any): Unit = items += item
  def seed(): Unit = add(Factory.makeCircle(1.0))
}

object RegistryFactory {
  def empty(): Registry = new Registry()
}
